src/metabase/query_processor/middleware/expand_macros.clj

```diff
@@ -1,206 +1,193 @@
 (ns metabase.query-processor.middleware.expand-macros
-  "Middleware for expanding `:metric` and `:segment` 'macros' in *unexpanded* MBQL queries.
```

Comment:
I completely rewrote this middleware, with the MLv2 tooling and `lib.walk` the new code is a lot easier to follow IMO.